extensions/notebook/src/dialog/managePackages/managePackagesDialogModel.ts

```
@@ -0,0 +1,204 @@
+			let provider = this._packageManageProviders.get(this.currentProvider);
+			return provider;
+		}
+		return undefined;
+	}
+
+	/**
+	 * Returns true if multi locations mode is enabled
+	 */
+	public get multiLocationMode(): boolean {
+		return this._options.multiLocations;
+	}
+
+	/**
+	 * returns the array of target locations
+	 */
+	public targetLocations(): string[] {
+		return Array.from(this._locations.keys());
+	}
+
+	/**
+	 * Returns the default location
+	 */
+	public get defaultLocation(): string {
+		return this._options.defaultLocation || this.targetLocations()[0];
+	}
+
+	/**
+	 * Loads the provider cache
+	 */
+	private async loadCaches(): Promise<void> {
+		if (this.packageManageProviders) {
+			let keyArray = Array.from(this.packageManageProviders.keys());
+			for (let index = 0; index < keyArray.length; index++) {
+				const element = this.packageManageProviders.get(keyArray[index]);
+				if (await element.canUseProvider()) {
+					if (!this._locations.has(element.packageTarget.location)) {
+						this._locations.add(element.packageTarget.location);
+					}
+					if (!this._packageTypes.has(element.packageTarget.location)) {
+						this._packageTypes.set(element.packageTarget.location, []);
+					}
+					this._packageTypes.get(element.packageTarget.location).push(element);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Returns the package types for given location
+	 */
+	public getPackageTypes(targetLocation?: string): [string, string][] {
+		targetLocation = targetLocation || this.defaultLocation;
+		let providers = this._packageTypes.get(targetLocation);
+		return providers.map(x => [x.providerId, x.packageTarget.packageType]);
+	}
+
+	/**
+	 * returns the list of packages for current provider
+	 */
+	public async listPackage(): Promise<IPackageDetails[]> {
+		let provider = this.currentPackageManageProvider;
+		if (provider) {
+			return await provider.listPackages();
+		} else {
+			return Promise.reject('Current Provider is not set');
+		}
```

You can just throw an Error here in async functions